addons/membership_sale/models/membership_sale.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from openerp import tools, models, fields, api
+
+class sale_order_line(models.Model):
+    _inherit = 'sale.order.line'
+
+    membership_start_date = fields.Date(
+        'Membership Start Date', default=lambda self: fields.date.today(), help='Date from which membership becomes active.')
+
+    @api.multi
+    def _create_membership_line(self):
+        for line in self:
+            if line.product_id.membership:
+                end_date = (datetime.strptime(line.membership_start_date, tools.DEFAULT_SERVER_DATE_FORMAT).date()) + relativedelta(months=line.product_id.membership_duration)
+                self.env['membership.membership_line'].create({
+                    'partner': line.order_partner_id.id,
+                    'membership_id': line.product_id.id,
+                    'member_price': line.price_unit,
+                    'date_from': line.membership_start_date,
+                    'date_to': end_date,
+                    'state': 'waiting',
+                    'sale_order_line_id': line.id
+                })
+        return True
+
+    @api.one
+    def button_confirm(self):
+        res = super(sale_order_line, self).button_confirm()
+        self._create_membership_line()
```

Comment:
need to re-factor?? write function prepare membership line and than create membership line